src/client/opamSolution.ml

```diff
@@ -1128,37 +1129,38 @@ let get_depexts ?(force=false) ?(recover=false) t packages =
             (* dirty heuristic: recompute for all non-canonical packages *)
             OpamPackage.Map.find_opt nv t.repos_package_index
             <> OpamSwitchState.opam_opt t nv)
-          packages
+          new_packages
       in
       if OpamPackage.Set.is_empty more_pkgs then base else
       OpamPackage.Map.union (fun _ x -> x) base
-        (OpamSwitchState.depexts_status_of_packages t more_pkgs)
+        (OpamSwitchState.depexts_status_of_packages t more_pkgs ~old_packages:(OpamPackage.Set.diff all_packages more_pkgs))
   in
-  let avail, nf =
-    OpamPackage.Set.fold (fun pkg (avail,nf) ->
+  let avail, required, nf =
+    OpamPackage.Set.fold (fun pkg (avail,req,nf) ->
         match OpamPackage.Map.find_opt pkg sys_packages with
         | Some sys ->
           OpamSysPkg.(Set.union avail sys.s_available),
+          OpamSysPkg.(Set.union req sys.s_required),
           OpamSysPkg.(Set.union nf sys.s_not_found)
-        | None -> avail, nf)
-      packages (OpamSysPkg.Set.empty, OpamSysPkg.Set.empty)
+        | None -> avail, req, nf)
```

Comment:
i think it might be worth using a dedicated record instead of a tuple at this point. Otherwise it could become confusing very quickly with all those similar types around

Reply:
We have `opamSysPkg.status`?

Reply:
Though, we're also passing around the couple of `available` and `required` a lot, which perhaps warrants a separate record.